### ISSUE TITLE
Inconsistency in expected values

### ISSUE THREAD

**Issue by rmelchiotti** [shap/shap#490]
Hi,
I am trying to use shap values to explain the results of a Gradient Boosting Classifier trained in sklearn.
![image](https://user-images.githubusercontent.com/28728698/54269834-01ca3f80-4576-11e9-8680-bf5e58117cc9.png).

I have difficulties understanding the results though. Why are the three expected values different? I can kind of understand why the first and second one would be different since the first one is agnostic of the training set but I don't get why there is any difference in the expected value between shap_values and shap_interaction_values. I tried repeating this using the boston dataset and in that case the expected value for shap_values and shap_interaction_values are the same.

Any clarification would be greatly appreciated

Thanks a lot for putting this package together, it is incredibly useful

**Comment by slundberg** [comment]
Wow, that's a bug. Thanks! Those should not have been different (all three should be the same). The second one is right, the first one is missing the base_offset, and the last one has a separate issue. I just fixed all this and pushed v0.28.6.

**Comment by rmelchiotti** [comment]
Thanks a lot!
